### PR TITLE
Fix/174 need versioned downloads

### LINKED ISSUE #174: Need Versioned Downloads
Long term only having the latest download is not acceptable. Need to be able to version lock ie ```wget https://spaceuptech.com/downloads/linux/space-cloud-0.6.7.zip```

edit: Better yet yarn add space-cloud@0.6.7

### PR DESCRIPTION
**Pull request by ledex**
Add version to deployment so that all build versions are stored. The version is determined from the ./version.go. This fixes #174.

### CODE REVIEW

**Comment by ledex** [comment]
I am not sure if this is the best way to determine the version. I can be exploited pretty easily. 
Does anyone have a better idea?

**Comment by YourTechBud** [comment]
We can use the `./space-cloud -v` to retrieve the version. Also, i think we should keep a plain `space-cloud.zip` file along with the versioned one. So we will be uploading 6 binaries.

**Comment by ledex** [comment]
@YourTechBud okay. I have changed the way the version is determined and also added the three plain ```space-cloud.zip``` files.

**Comment by YourTechBud** [comment]
@ledex seems perfect to me. Let's just squash all commits into one?

@Jayesh333 You should probably have a look at it and probably test it out. You were the one to set up travis after all.

**Comment by ledex** [comment]
@YourTechBud everything is squashed

**Comment by YourTechBud** [comment]
@Jayesh333 test it out once. I think it looks good.

File: deploy.sh [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
-GOOS=linux GOARCH=amd64 go build -ldflags '-s -w -extldflags "-static"' .
-zip space-cloud.zip space-cloud
-mv ./space-cloud.zip ./linux/
-rm space-cloud
 
 # echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
 # docker push spaceuptech/space-cloud:latest
 
+#Upload as versioned build
+curl -H "Authorization: Bearer $JWT_TOKEN" -F 'file=@./darwin/space-cloud_'$BUILD_VERSION'.zip' -F 'fileType=file' -F 'makeAll=false' -F 'path=/darwin' https://spaceuptech.com/v1/api/downloads/files
```
Comment (Jayesh333) [inline deploy.sh]: The single quotes over here are not required and might give an error:
```
'$BUILD_VERSION'.zip
```

**Comment by Jayesh333** [comment]
Also, we need to prefix the $BUILD_VERSION with `v`. So the final zip should look like `space-cloud_v$BUILD_VERSION`. 

It looks good overall. Make the requested changes and we are good to go! @ledex

File: deploy.sh (hunk 1 above)
Comment (ledex) [reply deploy.sh]: If the single quotes are removed like so 
```
curl -H "Authorization: Bearer $JWT_TOKEN" -F 'file=@./darwin/space-cloud_$BUILD_VERSION.zip' -F 'fileType=file' -F 'makeAll=false' -F 'path=/darwin' https://spaceuptech.com/v1/api/downloads/files

```
the variable is not processed and the file `space-cloud_$BUILD_VERSION.zip` is being looked for instead.

File: deploy.sh [hunk 2]
```diff
@@ -1,22 +1,31 @@
 #!/bin/bash
 
+#Build linux-version first and determine the build-version
+GOOS=linux GOARCH=amd64 go build -ldflags '-s -w -extldflags "-static"' .
+BUILD_VERSION=$(space-cloud -v | cut -f3 -d ' ')
+
+zip space-cloud_$BUILD_VERSION.zip space-cloud
+mv ./space-cloud_$BUILD_VERSION.zip ./linux/
```
[the remark this answers is not in the document]
Comment (ledex) [reply deploy.sh]: Put the folder-creation in front now.

File: deploy.sh (hunk 1 above)
Comment (ledex) [reply deploy.sh]: Okay. changed it to be in double-quotes completely.

**Comment by ledex** [comment]
Also squashed my changes ;) @YourTechBud

File: deploy.sh [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
 # echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
 # docker push spaceuptech/space-cloud:latest
 
+#Upload as versioned build
+curl -H "Authorization: Bearer $JWT_TOKEN" -F "file=@./darwin/space-cloud_v$BUILD_VERSION.zip" -F 'fileType=file' -F 'makeAll=false' -F 'path=/darwin' https://spaceuptech.com/v1/api/downloads/files
+curl -H "Authorization: Bearer $JWT_TOKEN" -F "file=@./windows/space-cloud_v$BUILD_VERSION.zip" -F 'fileType=file' -F 'makeAll=false' -F 'path=/windows' https://spaceuptech.com/v1/api/downloads/files
+curl -H "Authorization: Bearer $JWT_TOKEN" -F "file=@./linux/space-cloud_v$BUILD_VERSION.zip" -F 'fileType=file' -F 'makeAll=false' -F 'path=/linux' https://spaceuptech.com/v1/api/downloads/files
+
+#Upload as latest build
 curl -H "Authorization: Bearer $JWT_TOKEN" -F 'file=@./darwin/space-cloud.zip' -F 'fileType=file' -F 'makeAll=false' -F 'path=/darwin' https://spaceuptech.com/v1/api/downloads/files
```
Comment (Jayesh333) [inline deploy.sh]: `space-cloud.zip` does not exists. We should copy the `space-cloud_v$BUILD_VERSION.zip` as `space-cloud.zip` before this step.

**Comment by Jayesh333** [comment]
This is the last change @ledex

File: deploy.sh (hunk 3 above)
Comment (ledex) [reply deploy.sh]: Okey should finally be good to go :)

**Pull request merged**